Steer/DigitizerWorkflow/src/ITSMFTDigitWriterSpec.cxx

```diff
@@ -77,16 +77,18 @@ class ITSMFTDPLDigitWriter
     auto inDigits = pc.inputs().get<std::vector<o2::itsmft::Digit>>((detStr + "digits").c_str());
     auto inROFs = pc.inputs().get<std::vector<o2::itsmft::ROFRecord>>((detStr + "digitsROF").c_str());
     auto inMC2ROFs = pc.inputs().get<std::vector<o2::itsmft::MC2ROFRecord>>((detStr + "digitsMC2ROF").c_str());
-    auto inLabels = pc.inputs().get<MCCont*>((detStr + "digitsMCTR").c_str());
+    if (mWithMCTruth) {
```

Comment:
the better way would be to use the InputRecordWalker with a filter on the MC input, so you can avoid to pass the `withMCTruth` flag. The code is then simply adopting to the defined inputs.

Reply:
I need this in any case later on because I want to pass this information to the digitizer. (To  not calculate labels in the first place.)

Reply:
Sorry wrong location. In this case, it makes sense to use the walker indeed and forget about the member. Thanks.

Reply:
yes, but also in the digitizer you can get the information from the allowed outputs and pass the flag to the worker.

Reply:
Sorry. Could you provide me with code? Is there an "isAllowed" on inputs?

Reply:
check the TPCDigitRootWriterSpec around line 200 https://github.com/AliceO2Group/AliceO2/blob/92ec581da1bd18b2a914db7f48562991866c5f5a/Steer/DigitizerWorkflow/src/TPCDigitRootWriterSpec.cxx#L200
Bottom line is to iterator over inputs matching the filter. If the input route is not defined, there will be no entry in the iteration.

Reply:
I will leave it as it is for now. The code works and is readable.